api/src/modules/renown/kyc-service.ts

```diff
@@ -0,0 +1,201 @@
+
+export interface KYCServiceOptions {
+  // The URL of the Ceramic node
+  ceramicUrl?: string;
+  definition: RuntimeCompositeDefinition
+}
+
+export class KYCService {
+
+  protected client: ComposeClient;
+
+  protected eventSource: EventSource | null = null;
+
+  protected entries: any[] = [];
+
+  protected codec: Codec<any, any, any>;
+
+  constructor(options: KYCServiceOptions) {
+    const ceramic = options.ceramicUrl || "http://localhost:7007";
+    this.client = new ComposeClient({
+      ceramic,
+      definition: options.definition as RuntimeCompositeDefinition,
+    });
+
+    this.eventSource = new EventSource(`${ceramic}/api/v0/feed/aggregation/documents`)
+    this.codec = JsonAsString.pipe(AggregationDocument)
+  }
+
+  #fetchEntriesFromCeramic = async (first: number, skip: number) => {
+    const date = new Date().toISOString()
+    const entries = await this.client.executeQuery<{
+      verifiableCredentialEIP712Index: {
+        edges: { node: CeramicPowerhouseVerifiableCredential }[];
+      };
+    }>(fetchEntries, {
+      first,
+      skip,
+      input: {
+        and: [
+          {
+            or: [
+              {
+                where: {
+                  revocationDate: {
+                    greaterThan: date,
+                  },
+                },
+              },
+              {
+                where: {
+                  revocationDate: {
+                    isNull: true,
+                  },
+                },
+              },
+            ],
+          },
+        ],
+      },
+    });
+
+    return entries as any[];
+  }
+
+  #fetchEntryFromCeramic = async (issuerId: string, subjectId: string) => {
+    const date = new Date().toISOString()
+    const result = await this.client.executeQuery<{
+      verifiableCredentialEIP712Index: {
+        edges: { node: CeramicPowerhouseVerifiableCredential }[];
+      };
+    }>(fetchEntries, {
+      input: {
+        and: [
+          {
+            where: {
+              issuerId: {
+                equalTo: issuerId,
+              },
+              subjectId: {
+                equalTo: subjectId,
+              },
+            },
+          },
+          {
+            or: [
+              {
+                where: {
+                  revocationDate: {
+                    greaterThan: date,
+                  },
+                },
+              },
+              {
+                where: {
+                  revocationDate: {
+                    isNull: true,
+                  },
+                },
+              },
+            ],
+          },
+        ],
+      },
+    });
+
+    if (result.errors) {
+      return null;
+    }
+
+    const { data } = result;
+    if (!data || data.verifiableCredentialEIP712Index.edges.length === 0) {
+      return null;
+    }
+
+    return data.verifiableCredentialEIP712Index.edges[0]!.node;
+  }
+
+  #updateEntries = async () => {
+    const first = 100;
```

Comment:
First should be 0 no? Or `this.entries.length`

Reply:
first means the first 100 entries
skip means to skip x entries. we start with 0 and add 100 if there are more pages